primitives/state-machine/src/lib.rs

```diff
@@ -1060,6 +1060,73 @@ mod tests {
 		);
 	}
 
+	#[test]
+	fn append_storage_works() {
```

Comment:
Can we get one more, where we kill the `key` in the prospective and call append after it?

Reply:
I got it already written, but I also write one with an existing value that is not a vec and I do not find satisfiable test.
The problem is that we would need to read the whole vec (my first test  expected a panic but it did not panic due to the fact that content was a correct size at first position: so it looks like unless we parse the whole thing, it is undefined behavior).

Reply:
I did not find suitable solution (unless parsing the whole vec but it defeat the point (except if we use a cache of this first parsing but it is a lot more work)).
So I did remove the test.
I think it is something that runtime should be responsible to avoid.